test/solution-template-validation-tests/test/mainTemplateTests.js

```diff
@@ -109,18 +103,33 @@ describe('template files - ', () => {
     });
 
     describe('resources tests - ', () => {
-        // Not required pending Jianping's PR
-        it('resourceGroup().location must not be used in the solution template except as a defaultValue for the location parameter', () => {
+        describe('resource location format tests and resourceGroup().location tests - ', () => {
+            var expectedLocation1 = '[parameters(\'location\')]';
+            var expectedLocation2 = '[variables(\'location\')]';
             templateFileJSONObjects.forEach(templateJSONObject => {
                 var templateObject = templateJSONObject.value;
                 templateObject.should.have.property('resources');
-                var locationString = '[resourceGroup().location]';
-                var message = 'in file:' + templateJSONObject.filename + ' should NOT have location set to [resourceGroup().location]';
-                Object.keys(templateObject.resources).forEach(resource => {
-                    var val = templateObject.resources[resource];
-                    if (val.location) {
-                        val.location.should.withMessage(getErrorMessage(val, templateJSONObject.filepath, message)).not.be.eql(locationString);
+                var resources = Object.keys(templateObject.resources).map(function(key) {
+                    return templateObject.resources[key];
+                });
+                // each resource location should be "location": "[parameters('location')]" or ""[variables('location')]""
+                it.each(resources, 'resource named %s resource location should be location: [parameters(\'location\')] or [variables(\'location\')]', ['name'], function(element, next) {
```

Comment:
could be more readable:
location value of resource %s should be either [parameters('location')] or [variables('location')]